contrib/PFAS_concentrations/PFAS_concentrations.py

```diff
@@ -0,0 +1,761 @@
+            (https://pydov.readthedocs.io/en/stable/query_attribute.html)
+        sort_by:
+            Sort on one or multiple attributes.
+            (https://pydov.readthedocs.io/en/stable/sort_limit.html)
+        max_features: int
+            Limit the number of WFS features you want to be returned.
+            (https://pydov.readthedocs.io/en/stable/sort_limit.html)
+
+        Returns
+        -------
+        The downloaded surface water data.
+        """
+        logger.info(f"Downloading surface water data")
+        data_wfs_VMM_sw = self.wfs_request(
+            'pfas:pfas_oppwater',
+            location, query, sort_by, max_features)
+        data_wfs_OVAM = self.wfs_request(
+            'pfas:pfas_analyseresultaten',
+            location, query, sort_by, max_features)
+
+        data_wfs_VMM_sw = data_wfs_VMM_sw.drop_duplicates(
+            subset=data_wfs_VMM_sw.columns)
+        data_wfs_OVAM = data_wfs_OVAM.drop_duplicates(
+            subset=data_wfs_OVAM.columns)
+        data_wfs_OVAM = data_wfs_OVAM[data_wfs_OVAM['medium'] == 'Oppervlaktewater']
+
+        data_wfs_VMM_sw_len = len(data_wfs_VMM_sw)
+        data_wfs_OVAM_len = len(data_wfs_OVAM)
+
+
+        nb_datapoints = {"Surface_water_VMM" : data_wfs_VMM_sw_len}
+        self.dictionary["nb_datapoints"][0].update(nb_datapoints)
+        nb_datapoints = {"Surface_water_OVAM": data_wfs_OVAM_len}
+        self.dictionary["nb_datapoints"][0].update(nb_datapoints)
+
+        return data_wfs_VMM_sw, data_wfs_OVAM
+
+    def waste_water(self, location, query, sort_by, max_features):
+        """
+        Download the waste water data.
+
+        Parameters
+        ----------
+        location:
+            Query on location.
+            (https://pydov.readthedocs.io/en/stable/query_location.html)
+        query:
+            Find data based on one or more of its attribute values.
+            (https://pydov.readthedocs.io/en/stable/query_attribute.html)
+        sort_by:
+            Sort on one or multiple attributes.
+            (https://pydov.readthedocs.io/en/stable/sort_limit.html)
+        max_features: int
+            Limit the number of WFS features you want to be returned.
+            (https://pydov.readthedocs.io/en/stable/sort_limit.html)
+
+        Returns
+        -------
+        The downloaded waste water data.
+        """
+        logger.info(f"Downloading waste water data")
+        data_wfs_VMM_ww = self.wfs_request(
+            'pfas:pfas_afvalwater',
+            location, query, sort_by, max_features)
+
+        data_wfs_VMM_ww = data_wfs_VMM_ww.drop_duplicates(
+            subset=data_wfs_VMM_ww.columns)
+
+        data_wfs_VMM_ww_len = len(data_wfs_VMM_ww)
+
+        nb_datapoints = {"Waste_water_VMM": data_wfs_VMM_ww_len}
+        self.dictionary["nb_datapoints"][0].update(nb_datapoints)
+
+        return data_wfs_VMM_ww
+
+    def main(self, medium, location=None, query=None, sort_by=None, max_features=None, save=False):
```

Comment:
I don't know how useful the `query` and `sort_by` parameters are here in the `main()`  function. Both require attribute fields, and as a result for this to work these attribute fields need to be common to all the mediums you are requesting. I'm not familiar enough with the datasets to know whether this is the case?

Reply:
You are correct. At the moment this is not a good option, as the different mediums can have different attribute names. This can be implemented later.